execution_environments/execution_environments_overview.rst

```diff
@@ -0,0 +1,148 @@
+
+Traditionally, application dependencies are installed by administrators on top of
+an operating system using packaging management tools like RPM or Python-pip.
+
+The major drawback of such approach is that an application might require versions
+of dependencies different from those provided by default.
+
+In case of Ansible, a typical installation consists of ansible-core and a set of Ansible collections.
+
+At present, there are more than a hundred collections included in the Ansible package and
+hundreds more are available on Ansible Galaxy and Automation Hub for manual installation.
+Many of them have dependencies for their plugins, modules, roles and playbooks they provide.
+
+The Ansible collections can depend on the following pieces of software and their versions:
+
+* ansible-core 
+* Python
+* Python packages
+* system packages
+* other Ansible collections
+
+The dependencies have to be installed and sometimes can conflict with each other.
+
+One way to **partly** resolve dependency issue is
+to use Python virtual environments on Ansible control node.
+However, applied to Ansible, it has drawbacks and natural limitations.
+
+Content separation
+------------------
+
+Typically users do not really need the whole Ansible package with that hundred collections.
+In most cases, and it is considered a good practice, you need to install only ``ansible-core``
+plus a set of collections for specific tasks, nothing more.
+
+Let's say, there is an Ansible control node or a tool like Ansible AWX/Controller used by several users.
+
+Each user works with a limited set of services and wants to use only ``ansible-core``
+and a corresponding set of Ansible collections automating those services.
+
+Each user is not interested in having someone else's content the users do not need including dependencies
+which can potentially brake their pipelines.
+They might also want to preserve their installation clean from unnecessary packages
+and files belonging their colleagues.
+
+Simply put, users might want their content be separated.
+
+Portability
+-----------
+
+An Ansible user writes content for Ansible locally and wants to leverage the container technology
+to make your automation runtimes portable, shareable and easily deployable to testing and production environments.
+
+What is Execution Environment
+=============================
+
+Ansible, as a software application, can run in a container, thus, it can benefit from containerization the same as most other applications.
+
+The Ansible automation execution environment (hereinafter, execution environment or EE) is a container image serving as Ansible control node.
+
+The EE image contains:
+
+* ansible-core
+* ansible-runner
+* none or more Ansible collections
+* Python
+* Python and system dependencies
+* custom user needs
+
+Tools you can use EEs with
+-------------------------
+
+You can use EEs with:
+
+* ansible-navigator
+* Ansible AWX
+* Automation controller
+* ansible-runner
```

Comment:
although maybe a bit more advanced, the ansible vscode extention along side the vscode container exention is another nice way to take adavantage of the EE https://code.visualstudio.com/docs/devcontainers/containers

Reply:
(in which case, the shell opens in the container and the tradtional ansible-playboo command can be used)